src/modules/EventList/index.tsx

```diff
@@ -665,11 +681,13 @@ export const EventList = ({
              * when there are thousands of company ids.
              */
             if (watchlistUsername) {
+                mergeState({ loadingWatchlist: 'initialized' });
                 const companyIds = (msg.data || []).map(
                     (i: InstrumentID) => Object.values(i)[0] as ValueOf<InstrumentID>
                 ) as string[];
                 const watchlistId = await upsertPrimaryWatchlist(companyIds, watchlistUsername);
                 mergeState({ watchlistId });
+                mergeState({ loadingWatchlist: 'started' });
```

Comment:
Could this be combined to a single `mergeState({ loadingWatchlist: 'started', watchlistId });`?

Reply:
great catch!